packages/dd-trace/test/util.spec.js

```diff
@@ -86,3 +86,177 @@ describe('generatePointerHash', () => {
     expect(hash1).to.equal('2b90dffc37ebc7bc610152c3dc72af9f')
   })
 })
+
+describe('encodeValue', () => {
+  describe('basic type handling', () => {
+    it('handles string (S) type correctly', () => {
+      const result = encodeValue({ S: 'hello world' })
+      expect(Buffer.isBuffer(result)).to.be.true
+      expect(result).to.deep.equal(Buffer.from('hello world'))
+    })
+
+    it('handles number (N) as string type correctly', () => {
+      const result = encodeValue({ N: '123.45' })
+      expect(Buffer.isBuffer(result)).to.be.true
+      expect(result).to.deep.equal(Buffer.from('123.45'))
+    })
+
+    it('handles number (N) as type string or number the same', () => {
+      const result1 = encodeValue({ N: 456.78 })
+      const result2 = encodeValue({ N: '456.78' })
+      expect(Buffer.isBuffer(result1)).to.be.true
+      expect(result1).to.deep.equal(result2)
+    })
+
+    it('handles binary (B) type correctly', () => {
+      const binaryData = Buffer.from([1, 2, 3])
+      const result = encodeValue({ B: binaryData })
+      expect(Buffer.isBuffer(result)).to.be.true
+      expect(result).to.deep.equal(binaryData)
+    })
+  })
+
+  describe('edge cases', () => {
```

Comment:
i'm worried that there's no way to tell an error from a value that happens to be an empty string or something

Reply:
Good catch, adding logs and returning undefined for these edge cases so we know when an error has occurred